src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -324,27 +337,52 @@ public static void requestWithClient(final RequestOptions requestOptions,
 
         final HttpContext httpContext = HttpClientContext.create();
 
-        final FutureCallback<HttpResponse> futureCallback = new FutureCallback<HttpResponse>() {
+        final Timer.Context timerContext = registry == null ? null : timer(registry, request).time();
```

Comment:
I had a go to see what this looked like: https://github.com/rlinehan/clj-http-client/pull/2